python/cuml/test/test_nearest_neighbors.py

```diff
@@ -389,41 +398,61 @@ def test_knn_graph(input_type, nrows, n_feats, p, k, metric, mode,
         assert isspmatrix_csr(sparse_cu)
 
 
-@pytest.mark.parametrize("metric", valid_metrics(cuml_algo="sparse"))
-@pytest.mark.parametrize('nrows', [1, 10, 35])
-@pytest.mark.parametrize('ncols', [10, 35])
-@pytest.mark.parametrize('density', [0.8])
-@pytest.mark.parametrize('n_neighbors', [1, 4])
-@pytest.mark.parametrize('batch_size_index', [10, 20000])
-@pytest.mark.parametrize('batch_size_query', [10, 20000])
-def test_nearest_neighbors_sparse(nrows, ncols,
-                                  density,
+@pytest.mark.parametrize("metric", valid_metrics_sparse())
+@pytest.mark.parametrize('shape', [(100, 100, 0.4), (100, 15000, 0.04)])
+@pytest.mark.parametrize('n_neighbors', [4])
+@pytest.mark.parametrize('batch_size_index', [40000])
+@pytest.mark.parametrize('batch_size_query', [50000])
+def test_nearest_neighbors_sparse(shape,
                                   metric,
                                   n_neighbors,
                                   batch_size_index,
                                   batch_size_query):
 
+    nrows, ncols, density = shape
+
     if nrows == 1 and n_neighbors > 1:
         return
 
     a = cp.sparse.random(nrows, ncols, format='csr', density=density,
-                         random_state=32)
+                         random_state=35)
+    b = cp.sparse.random(nrows, ncols, format='csr', density=density,
+                         random_state=38)
 
-    logger.set_level(logger.level_info)
-    nn = cuKNN(metric=metric, n_neighbors=n_neighbors, algorithm="brute",
+    if metric == 'jaccard':
+        a = a.astype('bool').astype('float32')
+        b = b.astype('bool').astype('float32')
+
+    logger.set_level(logger.level_debug)
+    nn = cuKNN(metric=metric, p=2.0, n_neighbors=n_neighbors,
+               algorithm="brute", output_type="numpy",
                verbose=logger.level_debug,
                algo_params={"batch_size_index": batch_size_index,
                             "batch_size_query": batch_size_query})
     nn.fit(a)
 
-    cuD, cuI = nn.kneighbors(a)
+    import time
+
+    start = time.time()
+    cuD, cuI = nn.kneighbors(b)
+    print("cuML took %s" % (time.time() - start))
 
-    sknn = skKNN(metric=metric, n_neighbors=n_neighbors,
+    if metric not in sklearn.neighbors.VALID_METRICS_SPARSE['brute']:
+        a = a.todense()
+        b = b.todense()
+
+    sknn = skKNN(metric=metric, p=2.0, n_neighbors=n_neighbors,
                  algorithm="brute", n_jobs=-1)
     sk_X = a.get()
     sknn.fit(sk_X)
 
-    skD, skI = sknn.kneighbors(sk_X)
+    start = time.time()
```

Comment:
remove the timing code?

Reply:
I've kept this in here up to now just because I've had to keep adding it back in as I've been making changes and testing timings. I'll remove it before merge for sure.